merlin-sdk/src/main/java/gov/nasa/jpl/aerie/merlin/protocol/types/ValueSchema.java

```diff
@@ -516,15 +516,13 @@ public Boolean onVariant(final List<Variant> variants) {
     });
   }
 
-  public static final class Variant {
-    /** The unique internal name of this variant. */
-    public final String key;
-    /** The user-facing presentation of this variant. */
-    public final String label;
-
-    public Variant(final String key, final String label) {
-      this.key = Objects.requireNonNull(key);
-      this.label = Objects.requireNonNull(label);
+  public static record Variant(
+      String key,  // The unique internal name of this variant
+      String label // The user-facing presentation of this variant
+  ) {
```

Comment:
I think I prefer the style you have here, but note that we could also use `@param` javadocs on the `Variant` itself:

```
/**
 * @param key The unique internal name of this variant
 * @param label The user-facing presentation of this variant
 */
public static record Variant(String key, String label)
```

Technically speaking, this is *the* way to associate javadocs to these fields, if we want IDEs to recognize them as such.